### PR TITLE
Save sync tips to DB

### PR DESCRIPTION
**Pull request by terencechain**
This PR implements when performing a state transition (ie process block), the latest sync tips can get saved in the DB

### CODE REVIEW

**Review by rauljordan — APPROVED** [review]
Very straightforward, thanks!

**Comment by terencechain** [comment]
Although there's no risk, but let's hold this up until we merge @potuz PRs first

File: beacon-chain/blockchain/process_block.go
Comment (potuz) [inline beacon-chain/blockchain/process_block.go:517]: Should we return an error here? tips should never be len 0

File: beacon-chain/forkchoice/protoarray/store.go
Comment (potuz) [inline beacon-chain/forkchoice/protoarray/store.go:47]: Doesn't this return the actual map therefore invalidating the Lock?

**Review by potuz — DISMISSED** [review]

**Review by potuz — APPROVED** [review]

**Pull request merged**
